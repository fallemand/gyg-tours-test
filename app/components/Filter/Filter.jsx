import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import Dropdown from '../Dropdown';
import './_filter.scss';

class Filter extends React.Component {
  constructor(props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleDropdownChange = this.handleDropdownChange.bind(this);
    this.handleQueryChange = this.handleQueryChange.bind(this);
  }

  handleSubmit(event) {
    event.preventDefault();
    const { onFilter, field, value } = this.props;
    onFilter({
      field,
      value,
    });
  }

  handleDropdownChange(query) {
    const { onChange, value } = this.props;
    onChange({ field: query, value });
  }

  handleQueryChange(event) {
    const { value } = event.target;
    const { onChange, field } = this.props;
    onChange({ field, value });
  }

  render() {
    const { className, fields, field, value } = this.props;
    return (
      <form onSubmit={this.handleSubmit} className={classnames(className, 'filter')}>
        <Dropdown
          name="filter"
          title="Filter"
          className="filter__field"
          value={field}
          options={fields}
          onChange={this.handleDropdownChange}
        />
        <input
          className="filter__value"
          placeholder="Filter Value"
          type="text"
          value={value}
          disabled={!field}
          onChange={this.handleQueryChange}
        />
      </form>
    );
  }
}

Filter.propTypes = {
  className: PropTypes.string,
  onFilter: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  field: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  fields: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  })).isRequired,
};

Filter.defaultProps = {
  className: null,
  value: '',
  field: '',
};

export default Filter;
