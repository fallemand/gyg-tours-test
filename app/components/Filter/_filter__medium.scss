@import '../../styles/variables';

.filter {
  width: 250px;

  .filter__field {
    margin: 5px 0;
    border-radius: 2px 0 0 2px;
    -webkit-appearance: none;
    width: 40%;
    height: $filter-height;
    min-width: inherit;
    font-size: $font-size-16;
  }

  .filter__value {
    margin: 5px 0 5px -1px;
    border-radius: 0 4px 4px 0;
    padding: 7px 10px 8px;
    -webkit-appearance: none;
    width: 60%;
    height: $filter-height;
    vertical-align: bottom;
    font-size: $font-size-16;
  }
}
