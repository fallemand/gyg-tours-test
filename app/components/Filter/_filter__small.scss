@import '../../styles/variables';

.filter {
  width: 100%;

  .filter__field {
    width: 40%;
    display: inline-block;
  }

  .filter__value {
    border-radius: 0 2px 2px 0;
    border: 1px solid $border-color;
    padding: 8px 10px 7px;
    margin-left: -1px;
    width: 60%;
    display: inline-block;
  }
}
