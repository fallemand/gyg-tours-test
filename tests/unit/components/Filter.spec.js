import React from 'react';
import Filter from '../../../app/components/Filter';

describe('Filter', () => {
  const props = {
    onFilter: () => {},
    onChange: () => {},
    field: '__VAL1__',
    fields: [
      { value: '__VAL1__', label: '__LAB1__' },
      { value: '__VAL2__', label: '__LAB2__', selected: true },
      { value: '__VAL1__', label: '__LAB1__', disabled: true },
    ],
  };

  it('render component', () => {
    const component = shallow(<Filter {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('render with className', () => {
    const component = shallow(<Filter className="__CLASSNAME__" {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });
});
