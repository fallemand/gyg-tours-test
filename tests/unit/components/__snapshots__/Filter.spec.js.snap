// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter render component 1`] = `
"<form onSubmit={[Function: bound handleSubmit]} className='filter'>
  <Dropdown name='filter' title='Filter' className='filter__field' value='__VAL1__' options={{...}} onChange={[Function: bound handleDropdownChange]} />
  <input className='filter__value' placeholder='Filter Value' type='text' value='' disabled={false} onChange={[Function: bound handleQueryChange]} />
</form>"
`;

exports[`Filter render with className 1`] = `
"<form onSubmit={[Function: bound handleSubmit]} className='__CLASSNAME__ filter'>
  <Dropdown name='filter' title='Filter' className='filter__field' value='__VAL1__' options={{...}} onChange={[Function: bound handleDropdownChange]} />
  <input className='filter__value' placeholder='Filter Value' type='text' value='' disabled={false} onChange={[Function: bound handleQueryChange]} />
</form>"
`;
